src/components/TimeCardPage/TimeSheet.tsx

```diff
@@ -97,7 +105,28 @@ export default function Page() {
         //TODO - Upload timesheet to DB 
     }
 
-
+    const renderWarning = () => {
```

Comment:
🤔 Could be worth memoizing, not a huge deal but I think you mentioned some concern about how often the timesheets rerender. (see useMemo or useCallback)